src/components/Verse/VerseText.tsx

```diff
@@ -37,9 +47,23 @@ const VerseText = ({ words }: VerseTextProps) => {
           [styles.verseTextSpaceBetween]: isReadingMode && !centerAlignPage,
         })}
       >
-        {words?.map((word) => (
-          <QuranWord key={word.location} word={word} font={quranReaderStyles.quranFont} />
-        ))}
+        {words?.map((word, index) => {
+          const highlight =
+            segments &&
+            // example: bismillahirrahmanirrahim, is detected as 4 words with chapterTypeName 'word'
+            // + 1 word with chapterTypeName 'end'. So 5 word in total. Need to check before doing shouldHighlight
+            word.charTypeName === 'word' &&
```

Comment:
You can use `CharType` enum for this check `CharType.Word`.

Reply:
thanks